packages/core/contracts/cross-chain-oracle/OracleHub.sol

```diff
@@ -0,0 +1,127 @@
+     * @param messenger ParentMessenger contract that sends messages to ChildMessenger on network with ID `chainId`.
+     * @dev Only callable by the owner (presumably the Ethereum Governor contract).
+     */
+    function setMessenger(uint256 chainId, ParentMessengerInterface messenger) public nonReentrant() onlyOwner {
+        require(address(messenger) != address(0), "Invalid messenger contract");
+        messengers[chainId] = messenger;
+        emit SetParentMessenger(chainId, address(messenger));
+    }
+
+    /**
+     * @notice Publishes a DVM resolved price to the OracleSpoke deployed on the network linked  with `chainId`, or
+     * reverts if not resolved yet. This contract must be registered with the DVM to query price requests.
+     * The DVM price resolution is communicated to the OracleSpoke via the Parent-->Child messenger channel.
+     * @dev This method will return silently if already called for this price request, but will still attempt to call
+     * `messenger.sendMessageToChild` again even if it is a duplicate call. Therefore the Messenger contract for this
+     * `chainId` should determine how to handle duplicate calls.
+     * @param chainId Network to resolve price for.
+     * @param identifier Identifier of price request to resolve.
+     * @param time Timestamp of price request to resolve.
+     * @param ancillaryData extra data of price request to resolve.
+     */
+    function publishPrice(
+        uint256 chainId,
+        bytes32 identifier,
+        uint256 time,
+        bytes memory ancillaryData
+    ) public nonReentrant() {
+        // `getPrice` will revert if there is no price.
+        int256 price = _getOracle().getPrice(identifier, time, ancillaryData);
+        _publishPrice(identifier, time, ancillaryData, price);
+        messengers[chainId].sendMessageToChild(abi.encode(identifier, time, ancillaryData, price));
+    }
+
+    /**
+     * @notice Submits a price request originating from an OracleSpoke. Request data must be sent via the
+     * Child --> Parent Messenger communication channel. Returns silently if price request is a duplicate.
+     * @dev This contract must be registered to submit price requests to the DVM. Only the ParentMessenger
+     * can call this method. If the original requester on the child chain wants to expedite the Child --> Parent
+     * message, then they can call `requestPrice` on this contract for the same unique price request.
+     * @param data ABI encoded params with which to call `_requestPrice`.
+     */
+    function processMessageFromChild(uint256 chainid, bytes memory data) public nonReentrant() onlyMessenger(chainid) {
+        (bytes32 identifier, uint256 time, bytes memory ancillaryData) = abi.decode(data, (bytes32, uint256, bytes));
+        bool newPriceRequested = _requestPrice(identifier, time, ancillaryData);
+        if (newPriceRequested) {
+            _getOracle().requestPrice(identifier, time, ancillaryData);
+        }
```

Comment:
```suggestion
        bool newPriceRequested = _requestPrice(identifier, time, ancillaryData);
        if (_requestPrice(identifier, time, ancillaryData))  _getOracle().requestPrice(identifier, time, ancillaryData);
```

Reply:
Hm I think this is less readable the way you have it. It's not clear what the return value of `_requestPrice` is otherwise

Reply:
fair.